cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerServiceTest.java

```diff
@@ -165,6 +168,41 @@ public void testSimpleSchedulerLifecycle() throws Exception {
     checkState(Scheduler.ScheduleState.NOT_FOUND, scheduleIds);
   }
 
+  @Test
+  public void testScheduleUpdate() throws Exception {
+    AppFabricTestHelper.deployApplication(namespace, AppWithWorkflow.class);
+    ApplicationSpecification applicationSpecification = store.getApplication(appId);
+
+    schedulerService.schedule(program, programType, ImmutableList.of(timeSchedule1));
+    applicationSpecification = createNewSpecification(applicationSpecification, program, programType, timeSchedule1);
+    store.addApplication(appId, applicationSpecification, locationFactory.create("app"));
+    List<String> scheduleIds = schedulerService.getScheduleIds(program, programType);
+    Assert.assertEquals(1, scheduleIds.size());
+    checkState(Scheduler.ScheduleState.SUSPENDED, scheduleIds);
```

Comment:
Yes, in unit test we have lazy hack in which the triggers are paused when they are initially deployed. 
This PR gets rid of the lazy hack which we have in unit tests: PR https://github.com/caskdata/cdap/pull/3451